src/NzbDrone.Core/Notifications/Join/JoinProxy.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.Linq;
+using System.Net;
+using FluentValidation.Results;
+using NLog;
+using RestSharp;
+using NzbDrone.Core.Rest;
+using NzbDrone.Common.Serializer;
+
+namespace NzbDrone.Core.Notifications.Join
+{
+    public interface IJoinProxy
+    {
+        void SendNotification(string title, string message, JoinSettings settings);
+        ValidationFailure Test(JoinSettings settings);
+    }
+
+    public class JoinProxy : IJoinProxy
+    {
+        private readonly Logger _logger;
+        private const string URL = "https://joinjoaomgcd.appspot.com/_ah/api/messaging/v1/sendPush?";
+
+        public JoinProxy(Logger logger)
+        {
+            _logger = logger;
+        }
+
+        public void SendNotification(string title, string message, JoinSettings settings)
+        {
+            var request = new RestRequest(Method.GET);
+
+            try
+            {
+                SendNotification(title, message, request, settings);
+            }
+            catch (JoinException ex)
+            {
+                _logger.Error(ex, "Unable to send Join message.");
+                throw new JoinException("Unable to send Join notifications. " + ex.Message);
+            }
+        }
+
+        public ValidationFailure Test(JoinSettings settings)
+        {
+
+            const string title = "Test Notification";
+            const string body = "This is a test message from Sonarr.";
+            try
+            {
+                SendNotification(title, body, settings);
+                return null;
+            }
+            catch (JoinException ex)
+            {
+                _logger.Error(ex, "Unable to send test Join message.");
+                return new ValidationFailure("APIKey", ex.Message);
+            }
+        }
+
+        private void SendNotification(string title, string message, RestRequest request, JoinSettings settings)
+        {
+            try
+            {
+                var client = RestClientFactory.BuildClient(URL);
+
+                request.AddParameter("deviceId", "group.all");
+                request.AddParameter("apikey", settings.APIKey);
+                request.AddParameter("title", "[Sonarr] " + title);
+                request.AddParameter("text", message);
+                request.AddParameter("icon", "https://avatars3.githubusercontent.com/u/1082903"); // Use the Sonarr logo.
```

Comment:
@thezoggy : The Join API doesn't support sending a local file using Data URI or anything similar.  The only thing they support is a URL to an icon to display.

But, your comment did make me realize I wasn't using the Github CDN.  So I updated the PR to use the rawgit CDN.